sdk/storage/storage-file-share/src/ShareClient.ts

```diff
@@ -988,12 +988,22 @@ export class ShareClient extends StorageClient {
       };
 
       for (const identifier of response) {
+        let accessPolicy: any = undefined;
```

Comment:
Is there a Type you can use here instead of any?

Reply:
Thanks for the question. Please see [here](https://docs.microsoft.com/en-us/rest/api/storageservices/set-share-acl#remarks). From the **Establishing Share-Level Access Policies** section

> A stored access policy can specify the start time, expiry time, and permissions for the shared access signatures with which it is associated. Depending on how you want to control access to your share or file resource, you can specify all of these parameters within the stored access policy, and omit them from the URL for the shared access signature. Doing so permits you to modify the associated signature's behavior at any time, as well as to revoke it. Or you can specify one or more of the access policy parameters within the stored access policy, and the others on the URL. Finally, you can specify all of the parameters on the URL.

Based on this (and my past experience working with Storage REST API), only thing that is required when defining an access policy is the identifier. All other parameters are optional.

Regarding your 2nd question about using a Type instead of any, I simply copied the code from Storage Queue SDK (https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-queue/src/QueueClient.ts#L934) as there the implementation is correct :)

Reply:
It's a pity we don't have a proper interface defined for this.
```ts
{
    /**
     * @member {Date} startsOn the date-time the policy is active.
     */
    startsOn?: Date;
    /**
     * @member {string} expiresOn the date-time the policy expires.
     */
    expiresOn?: Date;
    /**
     * @member {string} permissions the permissions for the acl policy
     * @see https://docs.microsoft.com/en-us/rest/api/storageservices/set-share-acl
     */
    permissions: string;
  };
```

The  `AccessPolicy ` from underlying use `string` for `Date`.
```ts
export interface AccessPolicy {
  /**
   * The date-time the policy is active.
   * **NOTE: This entity will be treated as a string instead of a Date because the API can
   * potentially deal with a higher precision value than what is supported by JavaScript.**
   */
  startsOn?: string;
  /**
   * The date-time the policy expires.
   * **NOTE: This entity will be treated as a string instead of a Date because the API can
   * potentially deal with a higher precision value than what is supported by JavaScript.**
   */
  expiresOn?: string;
  /**
   * The permissions for the ACL policy.
   */
  permissions?: string;
}
```